examples/webgpu_compute_prefix_sums.html

```diff
@@ -0,0 +1,520 @@
+			<br /> Reference implementations and resources by <a href="https://github.com/b0nes164/GPUPrefixSums/">b0nes164</a>
+			<br/>
+			<div id="left_side" style="
+				position: absolute;
+				top: 150px;
+				left: 0;
+				padding: 10px;
+				background: rgba( 0, 0, 0, 0.5 );
+				color: #fff;
+				font-family: monospace;
+				font-size: 12px;
+				line-height: 1.5;
+				pointer-events: none;
+				text-align: left;
+			"></div>
+			<div id="right_side" style="
+			position: absolute;
+			top: 150px;
+			right: 0;
+			padding: 10px;
+			background: rgba( 0, 0, 0, 0.5 );
+			color: #fff;
+			font-family: monospace;
+			font-size: 12px;
+			line-height: 1.5;
+			pointer-events: none;
+			text-align: left;
+		"></div>
+		</div>
+
+		<script type="importmap">
+			{
+				"imports": {
+					"three": "../build/three.webgpu.js",
+					"three/webgpu": "../build/three.webgpu.js",
+					"three/tsl": "../build/three.tsl.js",
+					"three/addons/": "./jsm/"
+				}
+			}
+		</script>
+
+		<script type="module">
+
+			import * as THREE from 'three';
+			import { If, vec3, Loop, uniform, instancedArray, uv, uint, float, modInt, Fn, vec2, uvec2, floor, instanceIndex, invocationLocalIndex, workgroupBarrier } from 'three/tsl';
+
+			import { GUI } from 'three/addons/libs/lil-gui.module.min.js';
+
+			// Max number of elements in each display.
+			const maxElements = 16384;
+
+			const effectController = {
+				// Sqr root of 16834
+				leftGridDim: uniform( Math.sqrt( maxElements ) ),
+				rightGridDim: uniform( Math.sqrt( maxElements ) ),
+				validate: uniform( 0 ),
+				// Algorithm being displayed
+				'Left Method Name': 'Fake',
+				'Right Method Name': 'Kogge-Stone',
+			};
+
+			const algorithms = [
+				'Sklansky (No Partition)',
+				'Sklansky',
+				'Kogge-Stone',
+				'Fake',
+				'Incorrect'
+			];
+
+			const gui = new GUI();
+			gui.add( effectController, 'Left Method Name', algorithms );
+			gui.add( effectController, 'Right Method Name', algorithms );
+
+			const timestamps = {
+				left_side: document.getElementById( 'left_side' ),
+				right_side: document.getElementById( 'right_side' )
+			};
+
+			// Initialize Two Views
+			init( false );
+			init( true );
+
+			// When forceGlobalSwap is true, force all valid local swaps to be global swaps.
+			async function init( rightSide ) {
+
+				// Create prefix sort data
+				const array = new Uint32Array( Array.from( { length: maxElements }, ( _, i ) => {
+
+					return i;
+
+				} ) );
+
+				const infoArray = new Uint32Array( 5 ).fill( 1 );
+
+				const currentElementsBuffer = instancedArray( array );
+				const infoBuffer = instancedArray( infoArray );
+
+				const genericInitMethod = ( currentElements ) => {
+
+					currentElements.element( instanceIndex ).assign( 1 );
+			
+				};
+
+				// Useful loops
+
+				const LoopThroughWorkgroup = ( startValue, callback ) => {
+
+					const WORKGROUP_SIZE = uint( 64 ).toVar( 'WORKGROUP_SIZE' );
```

Comment:
I'm not sure if it's accurate to call these variables unused? They're not accessed as nodes but the WGSL variable that is created by the .toVar() call is used within LoopThroughWorkgroup(). However, it might make more sense to simply define workgroup size within the increment string itself.